src/calculate_diff_evalue.jl

```diff
@@ -0,0 +1,12 @@
+function calculate_diff_evalue(feature_map_circuit::AbstractBlock, cost::AbstractBlock, var::AbstractBlock, N::Int, theta, x; mapping::FeatureMap)
+	diff_evalue = 0.0
+	for j in 1:N
+		diff_evalue_pos = real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, j, π/2; mapping)))
+
+		diff_evalue_neg = real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, j, -π/2; mapping)))
```

Comment:
This could also be DRY:

```suggestion
        _get_diff_evalue(v::Real) =  real.(expect(cost, zero_state(N) => new_circuit(feature_map_circuit, var, x, theta, N, j, v; mapping)))
	diff_evalue = 0.0
	for j in 1:N
		diff_evalue_pos = _get_diff_evalue(π/2)

		diff_evalue_neg = _get_diff_evalue(-π/2)
```

Reply:
Maybe find a better function name than I have. Something that would be consistent with my [other comment](https://github.com/SciML/QuantumNLDiffEq.jl/pull/1/files#r835889393).

Reply:
@elisno My above comment applies here too. 
If I feel that there could be changes to the parameter shift rule used here, I'd consider generalizing it. For now, this is good enough :)